internal/identity/identity.go

```diff
@@ -44,14 +55,13 @@ func (d *Driver) GetPluginCapabilities(ctx context.Context, req *csi.GetPluginCa
 }
 
 // Probe returns the health and readiness of the plugin.
-func (d *Driver) Probe(ctx context.Context, req *csi.ProbeRequest) (*csi.ProbeResponse, error) {
-	d.log.WithField("method", "probe").Info("check whether the plugin is ready")
-	d.readyMu.Lock()
-	defer d.readyMu.Unlock()
+func (i *Identity) Probe(ctx context.Context, req *csi.ProbeRequest) (*csi.ProbeResponse, error) {
+	i.log.WithField("method", "probe").Info("check whether the plugin is ready")
 
 	return &csi.ProbeResponse{
 		Ready: &wrappers.BoolValue{
-			Value: d.ready,
+			// TODO: should we check e.g. that network is available or that all the cli tools are available ?
```

Comment:
CLI tools check makes sense atleast. Perhaps network as well.

Reply:
@ajmyyra thanks for the review :+1:  We have `checkSystemRequirements` function in the `internal/filesystem` tests that we could probably use here. I'll create separate PR for that.